plugins/sdexFeed.go

```diff
@@ -0,0 +1,90 @@
+package plugins
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/interstellar/kelp/api"
+	"github.com/interstellar/kelp/model"
+	"github.com/interstellar/kelp/support/utils"
+	"github.com/stellar/go/clients/horizon"
+)
+
+// sdexFeed represents a pricefeed from the SDEX
+type sdexFeed struct {
+	sdex       *SDEX
+	assetBase  *horizon.Asset
+	assetQuote *horizon.Asset
+}
+
+// ensure that it implements PriceFeed
+var _ api.PriceFeed = &sdexFeed{}
+
+// newSDEXFeed creates a price feed from buysell's url fields
+func newSDEXFeed(url string) (*sdexFeed, error) {
+	urlParts := strings.Split(url, "/")
+
+	baseParts := strings.Split(urlParts[0], ":")
+	baseCode := baseParts[0]
+	baseIssuer := baseParts[1]
+	baseConvert, e := utils.ParseAsset(baseCode, baseIssuer)
+	if e != nil {
+		return nil, fmt.Errorf("unable to convert base asset url to sdex asset: %s", e)
+	}
+
+	quoteParts := strings.Split(urlParts[1], ":")
+	quoteCode := quoteParts[0]
+	quoteIssuer := quoteParts[1]
+	quoteConvert, e := utils.ParseAsset(quoteCode, quoteIssuer)
+	if e != nil {
+		return nil, fmt.Errorf("unable to convert quote asset url to sdex asset: %s", e)
+	}
+
+	tradingPair := &model.TradingPair{
+		Base:  model.Asset(utils.Asset2CodeString(*baseConvert)),
+		Quote: model.Asset(utils.Asset2CodeString(*quoteConvert)),
+	}
+
+	sdexAssetMap := map[model.Asset]horizon.Asset{
+		tradingPair.Base:  *baseConvert,
+		tradingPair.Quote: *quoteConvert,
+	}
+
+	feedSDEX := MakeSDEX(
```

Comment:
can call this `sdex` instead of `feedSDEX` to keep the names as terse as needed

Reply:
done